### PR TITLE
docs(firestore-bigquery-export) Add instructions, information, and references for schema-views script

### PR DESCRIPTION
**Pull request by rachelsaunders**
- add instructions file for setting up for and running the schema-views script
- add example queries file for use with the schema-views script
- add references to PRE/POSTINSTALL file for schema-views script
- tighten up existing import script section to align with new schema-views script section in PRE/POSTINSTALL files

Issue #139

### CODE REVIEW

File: firestore-bigquery-export/POSTINSTALL.md [hunk 1]
```diff
@@ -28,9 +30,10 @@ The `bigquery-mirror-test` document will disappear from the **latest view** and
 1.  You can check the changelogs of a single document with this query:
 
     ```  
-    SELECT * FROM `${param:PROJECT_ID}.${param:DATASET_ID}.${param:TABLE_ID}_raw_changelog` 
-       WHERE document_name = "bigquery-mirror-test" 
-       ORDER BY TIMESTAMP ASC  
+    SELECT *
```
[the remark this answers is not in the document]
Comment (jhuleatt) [reply firestore-bigquery-export/POSTINSTALL.md]: yep should be fine!

File: firestore-bigquery-export/POSTINSTALL.md [hunk 2]
```diff
@@ -48,13 +51,17 @@ Note that this extension only listens for _document_ changes in the collection,
...
-**Important:** Run the script over the entire collection _after_ installing this extension, otherwise all writes to your database during the import might be lost.
+**Important:** Run the import script over the entire collection _after_ installing this extension, otherwise all writes to your database during the import might be lost.
 
-You may pause and resume the script from the last batch at any point.
+Learn more about using the import script to [backfill your existing collection](https://github.com/firebase/extensions/blob/master/firestore-bigquery-export/guides/IMPORT_EXISTING_DOCUMENTS.md).
 
-Learn more about using this script to [backfill your existing collection](https://github.com/firebase/extensions/blob/master/firestore-bigquery-export/guides/IMPORT_EXISTING_DOCUMENTS.md).
+### _(Optional)_ Generate schema views
+
+After exporting data to BigQuery using this extension, you can run the schema-views script (provided by this extension) to create BigQuery views, based on JSON schema configuration files that you create, using BigQuery’s built-in JSON functions.
```
Comment (jhuleatt) [inline firestore-bigquery-export/POSTINSTALL.md]: Suggest rewording to include more "why". Something like:

> Once your data is in BigQuery, create views that make it easier to query relevant data. You only need to provide a JSON schema file that describes your data structure, and the included schema-views script will handle the rest. 
>
> Learn more about [generating schema views](https://github.com/firebase/extensions/blob/master/firestore-bigquery-export/guides/GENERATE_SCHEMA_VIEWS.md).

File: firestore-bigquery-export/guides/GENERATE_SCHEMA_VIEWS.md [hunk 3]
```diff
@@ -0,0 +1,374 @@
...
+[gcloud](https://cloud.google.com/sdk/gcloud/) CLI:
+
+```
+$ gcloud auth application-default login
+```
+
+Alternatively, you can
+[create and use a service account](https://cloud.google.com/docs/authentication/production#obtaining_and_providing_service_account_credentials_manually).
+This service account must be assigned a role that grants the permission of
+`bigquery.jobs.create`.
```
Comment (laurenzlong) [inline firestore-bigquery-export/guides/GENERATE_SCHEMA_VIEWS.md]: Easier to say that they have to give it the role of "BigQuery Job User", so they don't have to look up which role has "bigquery.jobs.create" permission. 
(Docs are at https://cloud.google.com/iam/docs/understanding-roles if you're curious)

Comment (rachelsaunders) [reply firestore-bigquery-export/guides/GENERATE_SCHEMA_VIEWS.md]: Is this the most granular role available?  It looks like it is. @jhuleatt 

If it is, then we can say,
"... a role that grants the permission `bigquery.jobs.create`, like the ["BigQuery Job User" role](https://cloud.google.com/iam/docs/understanding-roles#bigquery-roles).

File: firestore-bigquery-export/POSTINSTALL.md (hunk 2 above)
Comment (IanWyszynski) [reply firestore-bigquery-export/POSTINSTALL.md]: +1

File: firestore-bigquery-export/guides/GENERATE_SCHEMA_VIEWS.md [hunk 4]
```diff
@@ -0,0 +1,374 @@
...
+      <td>timestamp</td>
+      <td>TIMESTAMP</td>
+    </tr>
+    <tr>
+      <td>geopoint</td>
+      <td>GEOGRAPHY</td>
+    </tr>
+    <tr>
+      <td>null</td>
+      <td>NULL</td>
```
Comment (IanWyszynski) [inline firestore-bigquery-export/guides/GENERATE_SCHEMA_VIEWS.md]: We should probably also include Firestore `reference` types here. Their corresponding values are "STRING"s containing the path to the referenced document.

File: firestore-bigquery-export/guides/GENERATE_SCHEMA_VIEWS.md [hunk 5]
```diff
@@ -0,0 +1,374 @@
...
+Review [this example](https://github.com/firebase/extensions/blob/master/firestore-bigquery-export/guides/EXAMPLE_QUERIES.md) for querying an array.
+
+Cloud Firestore arrays are
+[unnested](https://cloud.google.com/bigquery/docs/reference/standard-sql/query-syntax#unnest),
+so all array fields of the document are
+[cross joined](https://cloud.google.com/bigquery/docs/reference/standard-sql/query-syntax#cross-join)
+in the output table. The view retains the member and offset columns, which are
+called `${FIRESTORE_ARRAY_NAME}_member` and `${FIRESTORE_ARRAY_NAME}_index`,
+respectively. To make querying easier, the view includes these two columns
+instead of the original `ARRAY` value field.
```
Comment (IanWyszynski) [inline firestore-bigquery-export/guides/GENERATE_SCHEMA_VIEWS.md]: It's probably worth pointing out that we don't support deeply nested arrays. Arrays are basically expanded to BigQuery `ARRAY<STRING>` where the contents the JSON-serialized contents of each member. We could expand this functionality in future work.

https://github.com/firebase/extensions/blob/next/firestore-bigquery-export/scripts/gen-schema-view/src/udf.ts#L69-L78

File: firestore-bigquery-export/guides/GENERATE_SCHEMA_VIEWS.md [hunk 6]
```diff
@@ -0,0 +1,374 @@
...
+
+1.  Run the schema-views script via `npx` by running the following command:
+
+    ```
+    $ npx @firebaseextensions/fs-bq-schema-views \
+      --non-interactive
+      --project=${param:PROJECT_ID}
+      --dataset=${param:DATASET_ID}
+      --table-name-prefix=${param:TABLE_ID}
+      --schema-files=./test_schema.json
```
Comment (jhuleatt) [inline firestore-bigquery-export/guides/GENERATE_SCHEMA_VIEWS.md]: To make this copy and pasteable, it needs a `\` before every newline.

```markdown
```shell
npx @firebaseextensions/fs-bq-schema-views \
      --non-interactive \
      --project=${param:PROJECT_ID} \
      --dataset=${param:DATASET_ID} \
      --table-name-prefix=${param:TABLE_ID} \
      --schema-files=./test_schema.json
```

File: firestore-bigquery-export/guides/GENERATE_SCHEMA_VIEWS.md (hunk 5 above)
Comment (jhuleatt) [reply firestore-bigquery-export/guides/GENERATE_SCHEMA_VIEWS.md]: To clarify, if I have an array that looks like this:

`[[1,2,3],[4,5,6]]`, it won't get double-unnested, it will just create two rows, one with `[1,2,3]` and one with `[4,5,6]`, correct?

File: firestore-bigquery-export/POSTINSTALL.md (hunk 2 above)
Comment (rachelsaunders) [reply firestore-bigquery-export/POSTINSTALL.md]: awesome - done! :-)

File: firestore-bigquery-export/guides/GENERATE_SCHEMA_VIEWS.md (hunk 3 above)
Comment (rachelsaunders) [reply firestore-bigquery-export/guides/GENERATE_SCHEMA_VIEWS.md]: done :-)

File: firestore-bigquery-export/guides/GENERATE_SCHEMA_VIEWS.md (hunk 6 above)
Comment (rachelsaunders) [reply firestore-bigquery-export/guides/GENERATE_SCHEMA_VIEWS.md]: done :-)

File: firestore-bigquery-export/guides/GENERATE_SCHEMA_VIEWS.md (hunk 5 above)
Comment (jhuleatt) [reply firestore-bigquery-export/guides/GENERATE_SCHEMA_VIEWS.md]: Per the [docs](https://firebase.google.com/docs/firestore/manage-data/data-types):

> An array cannot contain another array value as one of its elements.

Does that mean we don't need to worry about it?
